automation_tools/__init__.py

```diff
@@ -573,31 +573,68 @@ def install_puppet_scap_client():
     run('yum -y install puppet-foreman_scap_client', warn_only=True)
 
 
-def setup_foreman_discovery():
-    """Task to setup foreman discovery."""
-    # Install required packages to enable discovery plugin
+def setup_foreman_discovery(sat_version, upstream=False):
+    """Task to setup foreman discovery.
+
+    :param str sat_version: should contain satellite version e.g. 6.1, 6.2
+    :param bool upstream: Flag whether we run on upstream. Default: ``False``.
+    """
     packages = (
         'tfm-rubygem-foreman_discovery',
         'rubygem-smart_proxy_discovery',
-        'foreman-discovery-image',
         'tfm-rubygem-hammer_cli_foreman_discovery',
     )
-    run('yum install -y {0}'.format(' '.join(packages)))
+    admin_password = os.environ.get('ADMIN_PASSWORD', 'changeme')
 
-    for daemon in ('foreman', 'httpd', 'foreman-proxy'):
-        manage_daemon('restart', daemon)
+    if sat_version in ('6.1', '6.2'):
```

Comment:
You wanted to exclude upstream here instead of including upstream in next if (see next comment)
```if not upstream and sat_version in ('6.1', '6.2'):```